### DISCUSSION TITLE
It looks like Yarn 3 picks up .yarnrc.yml settings from parent folders. Is there a way to disable that?

### DISCUSSION THREAD

**Discussion by cakoose** [yarnpkg/berry#4556] [Q&A]
Using Yarn 3.2.1.

- For our main application, I have a "package.json" and ".yarnrc.yml" in the top level of our repo. The yarnrc has `nodeLinker: node-modules`.
- For a GitHub Action utility, I have a subfolder ".github/workflows/pr-status-bot" that has its own "package.json" and ".yarnrc.yml". That yarnrc doesn't specify `nodeLinker` because the default is PnP.

When I run `yarn install` from the ".github/workflows/pr-status-bot" folder, Yarn seems to pick up the `nodeLinker: node-modules` setting from the top-level yarnrc. (When I delete the top-level yarnrc, it uses PnP.)

This was a bit surprising. Is there any way to tell Yarn to not search for additional yarnrcs in parent folders, e.g. a `noInherit: true` setting?

My workaround is to put `nodeLinker: pnp` in the GitHub Action's yarnrc. Are there any other settings that might be unexpectedly inherited?

**Comment by ReDrUm** [comment]
I just discovered this too. I'm in the process of migrating a newly formed monorepo from individual sandboxed project directories to use a global one, and I had planned to do it incrementally, however the inheritance from the root `.yarnrc.yml` is unexpected. I would have assumed if Yarn found a `yarn.lock` file it would have considered that the suitable root and stopped traversing. Adding an option would be great to facilitate control in an interim state.

**Comment by arcanis** [comment]
For now that's the expected behaviour. In a future major we'll likely restrict that to only the project top-level & the home folder user config, but that's long-term.

**Reply by ReDrUm** [reply]
Fair enough. Any appetite for a config setting to control it in the meantime? I'd be happy to whip one up. Perhaps something like:

```` yaml
rcTraversalMode: 'root' | 'lockfile'
````

or

``` yaml
rcTraversalMode: 'root' | 'project'
```

Where `root` is the default existing behaviour.
